Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/aspect/EventAspectAdvice.java

```
@@ -47,123 +48,124 @@
-        + "@annotation(gov.hhs.fha.nhinc.aspect.OutboundProcessingEvent) || "
-        + "@annotation(gov.hhs.fha.nhinc.aspect.NwhinInvocationEvent)", throwing = "e")
-    public void failEvent(JoinPoint joinPoint, Throwable e) {
-        if (eventRecorder != null && eventRecorder.isRecordEventEnabled()) {
-            MethodSignature sig = (MethodSignature) joinPoint.getSignature();
-            Method method = sig.getMethod();
-
-            // Check if the method is annotated with @LogFailures. We dont want to log the event twice.
-            if (method.getAnnotation(LogFailures.class) != null) {
-                Logger log = LoggerFactory.getLogger(sig.getDeclaringType()); //NOSONAR
-                log.warn("Method {} has multiple logging aspects. Will only process LogFailures annotation... ", method.getName());
-            } else {
-                logFailure(joinPoint, e);
-            }
+    public Object logEvents(ProceedingJoinPoint joinPoint, EventAdviceDelegate delegate, Class<? extends EventDescriptionBuilder> beforeBuilder,
+        Class<? extends EventDescriptionBuilder> afterBuilder, String serviceType, String version) throws Throwable {
+        delegate.begin(joinPoint.getArgs(), serviceType, version, beforeBuilder);
+
+        Object value = null;
+        boolean hasFailure = false;
+        try {
+            value = joinPoint.proceed();
+        } catch (ErrorEventException e) {
+            // Log the failure, and handle the exception. Service should still return a value in this case
+            logFailure(joinPoint, e, serviceType, version);
+            value =  e.getReturnOverride();
+            hasFailure = true;
+        } catch (Throwable e) {
+            // We don't want to swallow the exception if its not an ErrorEventException.
+            // We also don't want the delegate to log the END_* event.
+            logFailure(joinPoint, e, serviceType, version);
+            throw e;
```

if this throw, we never hit the logic below

---

That is the point. If we swallow the exception, then there will be an END_* event logged into the database. Per the Event Logging Approach wiki page, we want either a BEGIN_* and a corresponding END_*, or a BEGIN_* and a MESSAGE_PROCESSING_FAILED event. If we swallow the exception and ignore it, we will have all 3: BEGIN_*, MESSAGE_PROCESSING_FAILED, and END_*